mcs/class/Mono.Debugger.Soft/Test/dtest.cs

```diff
@@ -2476,14 +2475,18 @@ Event step_out_await (string method, Event e)
 
 				break;
 			}
-		} finally {
-			try {
-				vm.Detach ();
-			} finally {
-				vm = null;
-			}
+		} catch (VMDisconnectedException exc) {
+			Assert.Fail ("Got VMDisconnectedException rather than crash event");
```

Comment:
It's better to let the exception go through in that case, so we know what's the root failure.

You should keep part of the previously existing code, while still adding what you have here, as followed:
```
		} finally {
			try {
				vm.Detach ();

			} catch (VMDisconnectedException) {
				// This is fine
			} finally {
				vm = null;
			}
		}